dataset/src/main/scala/frameless/functions/NonAggregateFunctions.scala

```diff
@@ -143,20 +161,32 @@ trait NonAggregateFunctions {
     *
     * apache/spark
     */
-  def concat[T](c1: AbstractTypedColumn[T, String],
-                rest: AbstractTypedColumn[T, String]*): c1.ThisType[T, String] =
-    c1.typed(untyped.concat((c1 +: rest).map(_.untyped): _*))
+  def concat[T](c1: TypedColumn[T, String], xs: TypedColumn[T, String]*): TypedColumn[T, String] =
```

Comment:
Looks much cleaner :D yeah, not worth the generality for sure. Now that we don't need the depended type, we can actually go back to having one vararg parameter.